staging/src/k8s.io/kubectl/pkg/cmd/annotate/annotate.go

```diff
@@ -189,8 +195,11 @@ func (o *AnnotateOptions) Complete(f cmdutil.Factory, cmd *cobra.Command, args [
 	if err != nil {
 		return err
 	}
-	o.PrintObj = func(obj runtime.Object, out io.Writer) error {
-		return printer.PrintObj(obj, out)
+	o.PrintObj = func(operation string) printers.ResourcePrinterFunc {
+		return func(obj runtime.Object, out io.Writer) error {
+			o.PrintFlags.NamePrintFlags.Operation = operation
```

Comment:
Please see this https://github.com/kubernetes/kubernetes/blob/41d86b2716426cd3f87598a62ebcf2bfd4b130f2/staging/src/k8s.io/kubectl/pkg/cmd/label/label.go#L191 related to dry run message